section2/crud.rb

```diff
@@ -2,22 +2,26 @@ module Crud
   require 'bcrypt'
   puts "Module CRUD activated"
 
-  def self.create_hash_digest(password)
+  # def self.create_hash_digest(password)
+  def create_hash_digest(password)
     BCrypt::Password.create(password)
   end
 
-  def self.verify_hash_digest(password)
+  # def self.verify_hash_digest(password)
+  def verify_hash_digest(password)
     BCrypt::Password.new(password)
   end
 
-  def self.create_secure_users(list_of_users)
+  # def self.create_secure_users(list_of_users)
+  def create_secure_users(list_of_users)
     list_of_users.each do |user_record|
       user_record[:password] = create_hash_digest(user_record[:password])
     end
     list_of_users
   end
 
-  def self.authenticate_user(username, password, list_of_users)
+  # def self.authenticate_user(username, password, list_of_users)
+  def authenticate_user(username, password, list_of_users)
     list_of_users.find { |user_record|
       user_record[:username] == username && user_record[:password] == password 
     } || "Credentials were not correct"
```

Comment:
```suggestion
  def authenticate_user(list_of_users = [])
    list_of_users.find { |user_record|
      user_record[:username] == self.username && user_record[:password] == self.password 
    } || "Credentials were not correct"
```

Reply:
`  def authenticate_user(list_of_users = [])`
May I ask for the list_of_users here, why is it require to do `= []`? Is it to handle for when no argument is passed??

Reply:
yes. and to handle if the arg value is nil